File: mssp/examples/016-partial-and-complete/src/SCL/policy.mjs
// Deployment policy. Which combiner this deployment runs, and what a run that
// did not finish means here.
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const here = path.dirname(fileURLToPath(import.meta.url));
export const POLICY = JSON.parse(fs.readFileSync(path.join(here, "policy.json"), "utf8"));

export const combinerName = () => POLICY.combiner;
export const isFatal = () => POLICY.on_partial === "fatal";
export const describe = () =>
  `${POLICY.deployment}: combine with ${POLICY.combiner}, a partial run is ${POLICY.on_partial}`;

File: mssp/examples/017-finished-is-not-complete/src/SCL/policy.py
"""Deployment policy: what a declared-incomplete source means here."""
import json
import pathlib

POLICY = json.loads((pathlib.Path(__file__).parent / "policy.json").read_text(encoding="utf-8"))


def is_fatal():
    return POLICY["a_declared_incomplete_source_is"] == "fatal"


def describe():
    return f'{POLICY["deployment"]}: a declared-incomplete source is {POLICY["a_declared_incomplete_source_is"]}'
